node/core/approval-voting/src/tests.rs

```diff
@@ -2259,10 +2272,91 @@ fn subsystem_process_wakeup_trigger_assignment_launch_approval() {
 			candidate_entry.approval_entry(&block_hash).unwrap().our_assignment().unwrap();
 		assert!(our_assignment.triggered());
 
+		// Handle the first assignment import
+		handle_double_assignment_import(&mut virtual_overseer, candidate_index).await;
+
+		assert!(overseer_recv(&mut virtual_overseer).timeout(TIMEOUT / 2).await.is_none());
+
 		virtual_overseer
 	});
 }
 
+/// Ensure that when two assignments are imported, only one triggers the Approval Checking work
+pub async fn handle_double_assignment_import(
+	virtual_overseer: &mut VirtualOverseer,
+	candidate_index: CandidateIndex,
+) {
+	assert_matches!(
+		overseer_recv(virtual_overseer).await,
+		AllMessages::ApprovalDistribution(ApprovalDistributionMessage::DistributeAssignment(
+			_,
+			c_index,
+		)) => {
+			assert_eq!(candidate_index, c_index);
+		}
+	);
+
+	recover_available_data(virtual_overseer).await;
+	fetch_validation_code(virtual_overseer).await;
+
+	let first_message = virtual_overseer.recv().await;
+	let second_message = virtual_overseer.recv().await;
+
+	for msg in vec![first_message, second_message].into_iter() {
```

Comment:
Does it mean the test doesn't care about the order of these two? Otherwise two explicit asserts are better.

Reply:
There is no way to guarantee the ordering as the two blocks are imported asynchronously, so we cannot ensure that one completes before the other starts up.